src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultRazorDocumentMappingService.cs

```diff
@@ -877,30 +877,47 @@ private async Task<TextDocumentEdit[]> RemapVersionedDocumentEditsAsync(TextDocu
 
     private TextEdit[] RemapTextEditsCore(Uri virtualDocumentUri, RazorCodeDocument codeDocument, TextEdit[] edits)
     {
-        if (_languageServerFeatureOptions.IsVirtualCSharpFile(virtualDocumentUri))
+        var generatedDocument = GetGeneratedDocumentFromVirtualDocumentUri(virtualDocumentUri, codeDocument);
+        if (generatedDocument is null)
         {
-            var remappedEdits = new List<TextEdit>();
-            for (var i = 0; i < edits.Length; i++)
-            {
-                var projectedRange = edits[i].Range;
-                if (!TryMapFromProjectedDocumentRange(codeDocument, projectedRange, out var originalRange))
-                {
-                    // Can't map range. Discard this edit.
-                    continue;
-                }
-
-                var edit = new TextEdit()
-                {
-                    Range = originalRange,
-                    NewText = edits[i].NewText
-                };
+            return edits;
+        }
 
-                remappedEdits.Add(edit);
+        var remappedEdits = new List<TextEdit>();
```

Comment:
Consider using a pooled list rather than instantiating a new temporary `List<T>`.

```suggestion
        using var _ = ListPool<TextEdit>.GetPooledObject(out var remappedEdits);
```

Reply:
I didn't change this, other than changing whitespace, so would rather leave this to a follow up, just in case PR causes some issue and needs reverting.